onnxruntime/core/providers/dml/DmlExecutionProvider/src/precomp.h

```diff
@@ -31,6 +31,8 @@
 #include "External/D3DX12/d3dx12.h"
 
 #include <DirectML.h>
+#include "..\..\..\..\include\onnxruntime\core\common\common.h"
```

Comment:
>..\..\..\..\ [](http://example.com/codeflow?start=10&length=12)

Ew, avoid. You sure it won't work without the "..\..\..\.."?

Reply:
Tried it early, without going all the upto include folder, compiler is not able to identify this header file, throwing error. This path requires going outside the inner `onnxruntime` folder, where as line 51 resides inner `onnxruntime `folder. That's why it might need.

Reply:
Then see how other files include it, but definitely don't keep it like this 😭. e.g. We see in onnxruntime\core\codegen\passes\weight_layout\weight_layout.h:

```
#pragma once
#include "core/codegen/common/common.h"
#include "core/codegen/common/registry.h"
#include "core/common/common.h"
```